core/src/main/java/io/github/tcdl/msb/impl/ObjectFactoryImpl.java

```diff
@@ -54,7 +56,25 @@ public Requester createRequester(String namespace, RequestOptions requestOptions
      */
     @Override
     public ResponderServer createResponderServer(String namespace,  MessageTemplate messageTemplate, ResponderServer.RequestHandler requestHandler) {
-        return ResponderServerImpl.create(namespace, messageTemplate, msbContext, requestHandler);
+        return ResponderServerImpl.create(namespace, messageTemplate, msbContext, requestHandler, null);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override public ResponderServer createResponderServer(String namespace, MessageTemplate messageTemplate, ResponderServer.RequestHandler requestHandler,
+            Class<? extends Payload> payloadClass) {
+        return ResponderServerImpl.create(namespace, messageTemplate, msbContext, requestHandler, payloadClass);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override public PayloadConverter getPayloadConverter() {
```

Comment:
Maybe we could stick to eager initialization and avoid concurrency issues altogether?